### PR TITLE
do not commit clean account code

### PR DESCRIPTION
**Pull request by ayrat555**
with https://github.com/mana-ethereum/mana/pull/629 we started committing all code that was read from the database.

But we should commit only code for new accounts.

### CODE REVIEW

File: apps/blockchain/lib/blockchain/account/repo/cache.ex [hunk 1]
```diff
@@ -129,7 +129,10 @@ defmodule Blockchain.Account.Repo.Cache do
     else
       state_with_account = Account.put_account(state, address, account)
 
-      if code, do: Account.put_code(state_with_account, address, code), else: state_with_account
+      case code do
```
Comment (InoMurko) [inline apps/blockchain/lib/blockchain/account/repo/cache.ex]: what if you wrote the commit_account_cache something like:
```
defp commit_account_cache({address, {:dirty, account = nil, code}}, state), do: Account.del_account(state, address)
defp commit_account_cache({address, {:dirty, account, code = {:dirty, code}}}, state) do 
  state_with_account = Account.put_account(state, address, account)
  Account.put_code(state_with_account, address, code)
end
defp commit_account_cache({address, {:dirty, account, code}}, state) do 
  Account.put_account(state, address, account)
end
```

File: apps/blockchain/lib/blockchain/account/repo.ex [hunk 2]
```diff
@@ -188,14 +192,20 @@ defmodule Blockchain.Account.Repo do
...
+            {status, account, {:clean, value_to_cache}}
+          )
 
         repo_with_cached_code = %{updated_repo | cache: updated_cache}
 
         {repo_with_cached_code, found_code}
 
       true ->
-        {updated_repo, {:ok, code}}
+        {_status, code} = code
```
Comment (InoMurko) [inline apps/blockchain/lib/blockchain/account/repo.ex]: this `cond` is getting rather big, any plans to start breaking it apart?

Comment (ayrat555) [reply apps/blockchain/lib/blockchain/account/repo.ex]: I moved code caching to separate functions

File: apps/blockchain/lib/blockchain/account/repo/cache.ex (hunk 1 above)
Comment (ayrat555) [reply apps/blockchain/lib/blockchain/account/repo/cache.ex]: @InoMurko done

**Pull request merged**